lib/modules/datasource/go/base.ts

```diff
@@ -154,17 +155,37 @@ export class BaseGoDatasource {
       const parsedUrl = URL.parse(goSourceUrl);
 
       // TODO: `parsedUrl.pathname` can be undefined
-      const packageName = trimLeadingSlash(`${parsedUrl.pathname}`);
+      let packageName = trimLeadingSlash(`${parsedUrl.pathname}`);
 
-      const registryUrl = `${parsedUrl.protocol}//${parsedUrl.host}`;
+      const endpoint = GlobalConfig.get('endpoint')!;
+
+      const endpointPrefix = regEx('https://[^/]*/(.*?/)(api/v4/?)?').exec(
+        endpoint,
+      );
+
+      if (endpointPrefix) {
+        packageName = packageName.replace(endpointPrefix[1], '');
+      }
+
+      const registryUrl =
+        endpoint?.replace(regEx('api/v4/?$'), '') ??
+        `${parsedUrl.protocol}//${parsedUrl.host}`;
```

Comment:
```suggestion
        endpoint prefix ? endpoint.replace(regEx('api/v4/?$'), '') :
        `${parsedUrl.protocol}//${parsedUrl.host}`;
```
only do the endpoint stuff when endpoint is a prefix of the current dep. endpoint is always set here.

Reply:
Done